library.rb

```diff
@@ -49,4 +52,102 @@ def list_rentals
       end
     end
   end
+
+  def parse_people_json(data_array)
+    people_data = []
+    rental_data = []
+    data_array.each do |data|
+      data.rentals.each do |rental|
+        rental_data.push({ Date: rental.date, BookId: rental.book.id, PersonId: rental.person.id })
+      end
+      case data.class.to_s
+      when 'Student'
+        people_data.push({ Class: data.class, Name: data.name, ID: data.id, Age: data.age, Rentals: rental_data })
+      when 'Teacher'
+        people_data.push({ Class: data.class, Name: data.name, ID: data.id, Age: data.age,
+                           Specialization: data.specialization, Rentals: rental_data })
+      end
+      rental_data = []
+    end
+    JSON.generate(people_data)
+  end
+
+  def parse_book_json(data_array)
+    book_data = []
+    rental_data = []
+    data_array.each do |data|
+      data.rentals.each do |rental|
+        rental_data.push({ Date: rental.date, BookId: rental.book.id, PersonId: rental.person.id })
+      end
+      book_data.push({ Title: data.title, Author: data.author, ID: data.id, Rentals: rental_data })
+      rental_data = []
+    end
+    JSON.generate(book_data)
+  end
+
+  def parse_rental_json(data_array)
+    rental_data = []
+    data_array.each do |data|
+      rental_data.push({ Date: data.date, BookId: data.book.id, PersonId: data.person.id })
+    end
+    JSON.generate(rental_data)
+  end
```

Comment:
- [ ] On exit from the app, the data should be saved (all the data: books, people, and rentals) into JSON files. One per type of data should be stored and named accordingly (e.g. rentals.json). 
To achieve this functionality, add something like this when a user exits the app.
```
File.open("rentals.json", "w") do |f|
      f.write JSON.generate(@rentals)
    end
```

Reply:
Thanks for the review, we had a bug and now it's fixed 👍

Reply:
Well resolved!